### ISSUE TITLE
Storeconfig command shouldn't overwrite the certificates on KV

### ISSUE THREAD

**Issue by carlosedp** [containous/traefik#3349]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

After changing Traefik's config, I've ran a K8s job to store my ConfigMap in an already existing config in Consul KV store. The problem is that not only the configuration got overwritten (expected and desired) but the certificates also got overwritten and cleaned.

I tried changed the path for the certificate to check if Traefik would only update the config part but the certificates also got overwritten. Instead of having the certificates on `traefik-external/acme/account` I tried changing to `traefik-external-certificates/acme/account` with same results.

### What did you expect to see?

Only the config should be changed, not the certificates.

### What did you see instead?

Everything got overwritten and my certificates lost.

### Output of `traefik version`: (_What version of Traefik are you using?_)

Traefik 1.6.1, Consul KV 1.1.0

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: traefik-conf-external
  namespace: kube-system
data:
  traefik.toml: |
    defaultEntryPoints = ["http","https"]
    debug = false
    logLevel = "INFO"

    #Config to redirect http to https
    [entryPoints]
      [entryPoints.http]
        address = ":80"
        compress = true
        [entryPoints.http.redirect]
        entryPoint = "https"
      [entryPoints.https]
        address = ":443"
        compress = true
        [entryPoints.https.tls]

    [api]
      [api.statistics]
        recentErrors = 10

    [kubernetes]
      # Only create ingresses where the object has traffic-type: external label
      labelselector = "traffic-type=external"

    [metrics]
      [metrics.prometheus]
      buckets=[0.1,0.3,1.2,5.0]
      entryPoint = "traefik"

    [ping]
      entryPoint = "http"

    [accessLog]

    [consul]
      endpoint = "consul-traefik-consul.kube-system.svc:8500"
      watch = true
      prefix = "traefik-external"

    [acme]
      email = "email@gmail.com"
      storage = "traefik-external-certificates/acme/account"
      acmeLogging = true
      entryPoint = "https"
      OnHostRule = true
      #caServer = "https://acme-v01.api.letsencrypt.org/directory"
      [acme.httpChallenge]
        entryPoint="http"

```

My Store job:

```yml
---
apiVersion: batch/v1
kind: Job
metadata:
  name: traefik-kv-store
spec:
  backoffLimit: 3
  activeDeadlineSeconds: 100
  template:
    metadata:
      name: traefik-kv-store
    spec:
      containers:
      - name: storeconfig
        image: traefik:v1.5
        imagePullPolicy: IfNotPresent
        args: [ "storeconfig", "-c", "/config/traefik.toml" ]
        volumeMounts:
        - name: config
          mountPath: /etc/traefik
          readOnly: true
      restartPolicy: Never
      volumes:
      - name: config
        configMap:
          name: traefik-conf-external
```

Job log:

```
Pod: traefik-kv-store-2f2d4

2018/05/18 15:09:33 Storing configuration: {"LifeCycle":{"RequestAcceptGraceTimeout":0,"GraceTimeOut":10000000000},"GraceTimeOut":0,"Debug":false,"CheckNewVersion":true,"SendAnonymousUsage":false,"AccessLogsFile":"","AccessLog":{"format":"common"},"TraefikLogsFile":"","TraefikLog":null,"Tracing":null,"LogLevel":"INFO","EntryPoints":{"http":{"Address":":80","TLS":null,"Redirect":{"entryPoint":"https"},"Auth":null,"WhitelistSourceRange":null,"WhiteList":null,"Compress":true,"ProxyProtocol":null,"ForwardedHeaders":null},"https":{"Address":":443","TLS":{"MinVersion":"","CipherSuites":null,"Certificates":null,"ClientCAFiles":null,"ClientCA":{"Files":null,"Optional":false}},"Redirect":null,"Auth":null,"WhitelistSourceRange":null,"WhiteList":null,"Compress":true,"ProxyProtocol":null,"ForwardedHeaders":null}},"Cluster":null,"Constraints":[],"ACME":{"Email":"carlosedp@gmail.com","Domains":null,"Storage":"traefik-external-certificates/acme/account","StorageFile":"","OnDemand":false,"OnHostRule":true,"CAServer":"","EntryPoint":"https","DNSChallenge":null,"HTTPChallenge":{"EntryPoint":"http"},"DNSProvider":"","DelayDontCheckDNS":0,"ACMELogging":true,"TLSConfig":null},"DefaultEntryPoints":["http","https"],"ProvidersThrottleDuration":2000000000,"MaxIdleConnsPerHost":200,"IdleTimeout":0,"InsecureSkipVerify":false,"RootCAs":null,"Retry":null,"HealthCheck":{"Interval":30000000000},"RespondingTimeouts":null,"ForwardingTimeouts":null,"AllowMinWeightZero":false,"Web":null,"Docker":null,"File":null,"Marathon":null,"Consul":{"Watch":true,"Filename":"","Constraints":[],"Trace":false,"TemplateVersion":0,"DebugLogGeneratedTemplate":false,"Endpoint":"consul-traefik-consul.kube-system.svc:8500","Prefix":"traefik-external","TLS":null,"Username":"","Password":""},"ConsulCatalog":null,"Etcd":null,"Zookeeper":null,"Boltdb":null,"Kubernetes":{"Watch":true,"Filename":"","Constraints":[],"Trace":false,"TemplateVersion":0,"DebugLogGeneratedTemplate":false,"Endpoint":"","Token":"","CertAuthFilePath":"","DisablePassHostHeaders":false,"EnablePassTLSCert":false,"Namespaces":null,"LabelSelector":"traffic-type=external","IngressClass":""},"Mesos":null,"Eureka":null,"ECS":null,"Rancher":null,"DynamoDB":null,"ServiceFabric":null,"Rest":null,"API":{"EntryPoint":"traefik","Dashboard":true,"Debug":false,"CurrentConfigurations":null,"Statistics":{"RecentErrors":10}},"Metrics":{"Prometheus":{"Buckets":[0.1,0.3,1.2,5],"EntryPoint":"traefik"},"Datadog":null,"StatsD":null,"InfluxDB":null},"Ping":{"EntryPoint":"http"}}
```

Before running the job:

![keys](https://user-images.githubusercontent.com/20382/40242888-8436527a-5a84-11e8-9c90-6e0435423176.PNG)

After running the job:

![image](https://user-images.githubusercontent.com/20382/40242923-99c5a0be-5a84-11e8-8dc3-fb9e9fe7db43.png)

**Comment by dtomcej** [comment]
This particular issue has been addressed in https://github.com/containous/traefik/pull/3202. Can you test with the `containous/traefik: experimental` image to see if the storeconfig command works properly?
Thanks!

**Comment by carlosedp** [comment]
Can't test it since `containous/traefik:experimental` is not an ARM64 image.

**Comment by dtomcej** [comment]
Hello @carlosedp Did you want me to build you a custom image (ARM64) for you to test with? I can put it on my personal hub account if you would like to test it out.

**Comment by carlosedp** [comment]
@dtomcej yes, would be great. I can test in a couple of minutes.

**Comment by dtomcej** [comment]
Hello @carlosedp, I have pushed the image to `dtomcej/traefik:master-arm64-3c8c5eb`. Feel free to test out that image (it is based off of the master branch, so it will reflect code in the 1.7 release).

**Comment by dtomcej** [comment]
So does that mean that this issue is resolved? (albeit with the upcoming release)?

Thanks!

**Comment by carlosedp** [comment]
IMO yes, I've just used this image on the Job pod, not on the running Traefik pods but on what I see it's pretty fine. The plan is to add to 1.6.x or 1.7?

**Comment by dtomcej** [comment]
This code was merged already, and will be available in the upcoming 1.7 release.

**Comment by dtomcej** [comment]
Since this is marked as working, and has already been merged, I am going to go ahead and close this ticket!
